services/workflows-service/src/collection-flow/helpers/find-document-definition-by-type-and-category.ts

```diff
@@ -0,0 +1,32 @@
+import { IFormElement } from '@/common/ui-definition-parse-utils/types';
+
+export const findDocumentDefinitionByTypeAndCategory = (
+  elements: Array<IFormElement<any>>,
+  { type, category }: { type: string; category: string },
+) => {
+  let documentDefinition: IFormElement<any> | null = null;
+
+  const findDocumentDefinitionRecursively = (
+    elements: Array<IFormElement<any>>,
+  ): IFormElement<any> | null => {
+    for (const element of elements) {
+      if (element.element === 'documentfield') {
+        if (
+          element.params?.template?.type === type &&
+          element.params?.template?.category === category
+        ) {
+          documentDefinition = element;
+          break;
+        }
+      }
```

Comment:
Can also use Array.prototype.every to help readability.
```suggestion
      if (
      element.element === 'documentfield' &&
      element.params?.template?.type === type &&
       element.params?.template?.category === category
       ) {
          documentDefinition = element;
          
          break;
      }
```

Reply:
Applied snippet, thanks.